### ISSUE TITLE
shimv2: Improve debuggability

### ISSUE THREAD

**Issue by fidencio** [kata-containers/kata-containers#366]
**Which feature do you think can be improved?**

Currently `containerd-shim-kata-v2` binary has as an option `-debug`. On CRI-O side, and possibly on contanerd as well, `-debug` is passed down when calling the binary whenever CRI-O log level is set to `debug`.

**How can it be improved?**

What's described above is horrible as we either don't have enough debug info, or we have too much debug info.
CRI-O supports the following debug levels: fatal, panic, error, warn, info, debug and trace. Kata supports at least warn, info, and debug. Knowing that, would be really good to have a way to make the CRI-O (or containerd) log level match with the Kata one.

**Additional Information**

This can be achieved in different ways, such as:
- Adding `-info`, `-warn`, command line options;
- Deprecated `-debug` command line option, and adding a `--log-level` option;
- Using shimv2 CreateTaskRequest API, passing this value as part of the `Options` field;

I'd like to have the discussion here on what would be the preferred way to proceed with this, and also learn about possible limitations that I'm not considering.

**Comment by liubin** [comment]
How about add an argument in `configuration.toml`, this will not need to modify caller(e.g., `containerd`).

**Comment by marcel-apf** [comment]
Hi! I want to "handle" this task as a way to start to contributing to the Kata project. Thanks @fidencio that pointed me to it.

**Comment by bergwolf** [comment]
Please note that modifying `CreateTaskRequest` API may introduce backward incompatiblity.

**Comment by fidencio** [comment]
@bergwolf, sure, we're aiming for the getting @liubin's suggestion implemented.

**Comment by ariel-adam** [comment]
@marcel-apf Hi Marcel, how is it going with this issue :-) ?

**Comment by marcel-apf** [comment]
> @marcel-apf Hi Marcel, how is it going with this issue :-) ?

The issue is blocked by a prerequisite bug: https://github.com/kata-containers/runtime/issues/3039 that has an outstanding PR: https://github.com/kata-containers/runtime/pull/3040.

In the meantime v1 is not that relevant, so I am considering "porting" the PR to v2, I will contact @fidencio  for some assistance.

**Comment by cmaf** [comment]
@marcel-apf are there still plans for a PR, or has this been resolved possibly in PRs not linked to this particular issue? We can close it if it is no longer relevant.
